packages/template-retail-react-app/app/components/_app-config/index.jsx

```diff
@@ -84,10 +81,19 @@ AppConfig.restore = (locals = {}) => {
         typeof window === 'undefined'
             ? locals.originalUrl
             : `${window.location.pathname}${window.location.search}`
-    const site = resolveSiteFromUrl(path)
-    const locale = resolveLocaleFromUrl(path)
 
+    let site
+    let locale
     const {app: appConfig} = getConfig()
+
+    if (appConfig.useXForwardedHost && locals.xForwardedHost) {
+        site = resolveSiteFromUrl(path, locals.xForwardedHost)
```

Comment:
We need to pass the xForwardedHost from locals object into into `absoluteUrl` when useXForwardedHost flag is true to make sure it will use the correct origin. If not, calling getAppOrigin would suffice